src/main/java/com/teragrep/pth_06/ArchiveMicroStreamReader.java

```diff
@@ -252,6 +356,23 @@ public InputPartition[] planInputPartitions(Offset start, Offset end) {
                         );
             }
 
+            // HDFS tasks
```

Comment:
would it make sense to combine these 3 for-loops into one and add them all to a list in one iteration?

Reply:
Solved in commit 1176cf1ab6653b0784129c59585d5995d2f7c7e1